account_global_discount/models/account_invoice.py

```diff
@@ -0,0 +1,237 @@
+            currency_id = self.currency_id.with_context(date=self.date_invoice)
+            amount_total_company_signed = currency_id.compute(
+                self.amount_total, self.company_id.currency_id)
+            amount_untaxed_signed = currency_id.compute(
+                self.amount_untaxed, self.company_id.currency_id)
+        sign = self.type in ['in_refund', 'out_refund'] and -1 or 1
+        self.amount_total_company_signed = amount_total_company_signed * sign
+        self.amount_total_signed = self.amount_total * sign
+        self.amount_untaxed_signed = amount_untaxed_signed * sign
+
+    def get_taxes_values(self):
+        round_curr = self.currency_id.round
+        tax_grouped = super().get_taxes_values()
+        for key in tax_grouped.keys():
+            base = tax_grouped[key]['base']
+            amount = tax_grouped[key]['amount']
+            for discount in self.global_discount_ids:
+                base = discount._get_global_discount_vals(
+                    base)['base_discounted']
+                amount = discount._get_global_discount_vals(
+                    amount)['base_discounted']
+            tax_grouped[key]['base'] = round_curr(base)
+            tax_grouped[key]['amount'] = round_curr(amount)
+        return tax_grouped
+
+    @api.model
+    def invoice_line_move_line_get(self):
+        """Append global discounts move lines"""
+        res = super().invoice_line_move_line_get()
+        for discount in self.invoice_global_discount_ids:
+            if not discount.discount:
+                continue
+            res.append({
+                'invoice_global_discount_id': discount.id,
+                'global_discount_id': discount.global_discount_id.id,
+                'type': 'global_discount',
+                'name': discount.name,
+                'price_unit': discount.discount_amount * -1,
+                'quantity': 1,
+                'price': discount.discount_amount * -1,
+                'account_id': discount.account_id.id,
+                'account_analytic_id': discount.account_analytic_id.id,
+                'invoice_id': self.id,
+            })
+        return res
+
+
+class AccountInvoiceGlobalDiscount(models.Model):
+    _name = "account.invoice.global.discount"
+    _description = "Invoice Global Discount"
+
+    name = fields.Char(
+        string='Discount Name',
+        required=True,
+    )
+    invoice_id = fields.Many2one(
+        'account.invoice',
+        string='Invoice',
+        ondelete='cascade',
+        index=True,
+        readonly=True,
+    )
+    global_discount_id = fields.Many2one(
+        comodel_name='global.discount',
+        string='Global Discount',
+        readonly=True,
+    )
+    discount = fields.Float(
+        string='Discount',
+        readonly=True,
+    )
+    discount_display = fields.Char(
+        compute='_compute_discount_display',
+        readonly=True,
+        string="Discount",
```

Comment:
Just change this label to get rid of the runbot warning

Reply:
```suggestion
        string="Displayed Discount",
```